pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/housekeeping/handler/EmailNotificationHandler.java

```diff
@@ -89,6 +99,32 @@ public void handleMessage(NotificationMessage message) {
             }
         }
 
+        //skip emails for PlayWright generated users / Email deny list
+        if (!emailDenyPatterns.isEmpty()) {
+            boolean hasDenyListEmail = false;
+            List<String> nonDenyEmailList = new ArrayList<>();
+            for (String toAddress : message.getDistributionList()) {
+                boolean isDenyEmail = false;
+                Email toEmail = new Email(toAddress, toAddress);
+                for (String emailPattern : emailDenyPatterns) {
+                    Pattern pattern = Pattern.compile(emailPattern);
+                    Matcher matcher = pattern.matcher(toEmail.getEmail());
+                    if (matcher.find()) {
+                        isDenyEmail = true;
+                        hasDenyListEmail = true; //to handle email DL with a PW test user and other emails
+                        break;
+                    }
+                }
+                if (!isDenyEmail) {
+                    nonDenyEmailList.add(toEmail.getEmail());
+                }
+            }
+            if (hasDenyListEmail && nonDenyEmailList.isEmpty()) {
```

Comment:
Your comment above may be directed at this (I don't quite understand it), but I'm not sure why you need an address list, since you never use the contents. You might also get away with one flag and a break statement if part of this clause was in a separate method? But I may be missing something here.

Reply:
Refactored code to handle it little differently. added detailed comments explaining logic